src/dataset.cpp

```diff
@@ -633,6 +686,15 @@ DatasetSlice DatasetSlice::operator+=(const ConstDatasetSlice &other) const {
       [](VariableSlice &a, const ConstVariableSlice &b) { return a += b; },
       *this, other);
 }
+DatasetSlice DatasetSlice::operator+=(const Variable &other) const {
```

Comment:
Question: do we also need to do all these operators for `VariableSlice` as well?

Reply:
It would be efficient to do so. At the moment the fact they are convertible to `Variable` gives the feature. I would do this as a separate change set.

Reply:
ok it's perfectly fine if it works through conversion